### PR TITLE
Create/Update event subscriptions in Workflow::Step::BranchPackageStep

### PR DESCRIPTION
**Pull request by dmarcoux**
It belongs in this class since we only want to report back to SCMs whenever we branch a package. A workflow without this step won't report to the SCM. At least, not for now.

### CODE REVIEW

**Comment by dmarcoux** [comment]
Reason for the **DO NOT MERGE** label: Wait for #11219 to be merged, then this PR can be rebased and also merged.

**Comment by dmarcoux** [comment]
[Support sync events](https://trello.com/c/vfKme6b1/1673-support-sync-events)

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 1]
```diff
@@ -120,6 +122,19 @@ def branch_request_file_gitlab
           project: { http_url: @scm_extractor_payload[:http_url] },
           object_attributes: { source: { default_branch: @scm_extractor_payload[:commit_sha] } } }.to_json
       end
+
+      def set_or_update_subscriptions(package)
+        ['Event::BuildFail', 'Event::BuildSuccess'].each do |build_event|
+          subscription = EventSubscription.first_or_create!(eventtype: build_event,
+                                                            receiver_role: 'reader', # We pass a valid value, but we don't need this.
+                                                            user: User.session.login,
```
Comment (saraycp) [inline src/api/app/models/workflow/step/branch_package_step.rb:130]: Please don't forget to use `@token.user` here unless you plan to do it in another commit/PR.

And do the same in the `branch` method.

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 2]
```diff
@@ -120,6 +122,19 @@ def branch_request_file_gitlab
           project: { http_url: @scm_extractor_payload[:http_url] },
           object_attributes: { source: { default_branch: @scm_extractor_payload[:commit_sha] } } }.to_json
       end
+
+      def create_or_update_subscriptions(package)
+        ['Event::BuildFail', 'Event::BuildSuccess'].each do |build_event|
+          subscription = EventSubscription.first_or_create!(eventtype: build_event,
+                                                            receiver_role: 'reader', # We pass a valid value, but we don't need this.
+                                                            user: User.session.login,
```
Comment (vpereira) [inline src/api/app/models/workflow/step/branch_package_step.rb]: can it be anonymous? Otherwise `User.session!.login`

**Review by vpereira — CHANGES_REQUESTED** [review]

File: src/api/app/models/workflow/step/branch_package_step.rb (hunk 2 above)
Comment (dmarcoux) [reply src/api/app/models/workflow/step/branch_package_step.rb]: This diff is outdated. See the commit `Take user from the token, not from the session` and its description message for details.

**Review by saraycp — APPROVED** [review]

**Review by danidoni — APPROVED** [review]
